test_utils_helpers.py

```
@@ -0,0 +1,1451 @@
+                return value
+            call_count[0] += 1
+            return return_values[-1]
+        
+        result = retry_with_backoff(function_with_varying_returns, max_retries=3)
+        assert result == "success"
+    
+    def test_retry_timeout_simulation(self):
+        """Test retry with simulated timeout scenarios"""
+        import time
+        
+        start_time = time.time()
```

Flake8 (7.2.0)

[error] local variable 'start_time' is assigned to but never used

(F841)